File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int, int> umap;
        umap[target - nums[0]] = 0;

        for (int i = 1; i < nums.size(); ++i) {

            if (umap.find(nums[i]) != umap.end()) {
                res.push_back(umap[nums[i]]);
                res.push_back(i);
                break;
            }
            umap[target - nums[i]] = i;
        }
        return res;
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int[] result = new int[2];
        for(int i = 0; i< nums.length; ++i){
            if(mp.containsKey(nums[i])){
                result[0] = mp.get(nums[i]);
                result[1] = i;
                return result;
            }
            mp.put(target-nums[i],i);
        }
        return result;
        
    }
}